com.unity.ml-agents/Tests/Editor/Areas/TrainingAreaReplicatorTests.cs

```diff
@@ -0,0 +1,42 @@
+using System.Linq;
+using NUnit.Framework;
+using Unity.Mathematics;
+using Unity.MLAgents.Areas;
+using UnityEngine;
+
+namespace Unity.MLAgents.Tests.Areas
+{
+    [TestFixture]
+    public class TrainingAreaReplicatorTests
+    {
+        private TrainingAreaReplicator m_Replicator;
+
+        [SetUp]
+        public void Setup()
+        {
+            var gameObject = new GameObject();
+            var trainingArea = new GameObject();
+            trainingArea.name = "MyTrainingArea";
+            m_Replicator = gameObject.AddComponent<TrainingAreaReplicator>();
+            m_Replicator.numAreas = 10;
+            m_Replicator.baseArea = trainingArea;
+            m_Replicator.Awake();
+            m_Replicator.OnEnable();
+        }
+
+        [Test]
+        public void TestComputeGridSize()
+        {
+            Assert.GreaterOrEqual(m_Replicator.GridSize.x * m_Replicator.GridSize.y * m_Replicator.GridSize.z, m_Replicator.numAreas);
+            Assert.AreEqual(new int3(2, 2, 6), m_Replicator.GridSize);
+        }
```

Comment:
Can you test this for a few more values (e.g. [1,2,5,7,8]) so you hit small numbers as well as perfect cube and perfect cube - 1?